index.js

```diff
@@ -39,30 +40,100 @@ function getTorrentMetaBuffer () { // TODO Fix parsing bug so this can be remove
 }
 
 function getFile (fpath) {
-  return _getFile(this, fpath)
+  return _getFile(module.exports, fpath)
+  .then(file => {
+    debug('FILE path=' + (file || {}).path, 'found=' + (file != null), file)
+    return file
+  })
+}
+
+function fetch (req, opts) {
+  opts = opts || {}
+  let inject = false
+  let url = null
+
+  // Convert a FetchEvent to Request
+  if (global.FetchEvent && req instanceof global.FetchEvent) {
+    inject = req.clientId == null
+    req = req.request // req is now an instance of Request
+  }
+
+  // Convert a Request to an URL
+  if (req instanceof global.Request) {
+    url = new URL(req.url)
+    inject = inject && url.search.substr(1)
+             .split('&').find(s => s === 'noPlanktosInjection') == null
+    if (req.method !== 'GET') throw new Error('Only HTTP GET requests supported')
+  } else if (req instanceof URL) {
+    url = req
+  } else if (typeof req === 'string') {
+    url = new URL(req)
+  }
+
+  if (req == null) throw new Error('Must provide a FetchEvent, Request, URL, or a string')
```

Comment:
Since this is the last comment and it seems resolved, I'm going to go ahead an merge this into master so I can open a pr for the tab-elect changes without having to do this weird base branch thing. If you think this is still a problem we can fix it in master 👍